languages/elixir/exercises/concept/basic-strings/.meta/design.md

```diff
@@ -0,0 +1,51 @@
+# Design
+
+## Goal
+
+The goal of this exercise is to teach the student how strings are implemented and used in Elixir.
+
+## Learning objectives
+
+After completing this exercise, the student should:
+
+- Know how to concatenate strings.
+- Know how to manipulate strings.
+- Know how to perform string interpolation.
+- Know how to create multiline strings (heredocs).
+
+## Out of scope
+
+- Strings as binaries for binary pattern matching.
+- Regular expressions.
+- Sigils.
+- Charlists.
+
```

Comment:
I kept re-reading the String module documentation while doing this, it talks a lot about code points and graphemes, so I get the feeling like this might be an important topic to teach. On the other hand, I have been working with Elixir for 3 years and never had to worry about it much, so maybe it's too detailed for this exercise.

Should I explicitly mention that the concept of graphemes is out of scope?

Reply:
I think it could be a discussion point for the `after.md` document, but it also means that there might be another exercise which should occur after this which then maybe features a recursive or iterating through a string (akin to rna-transcription) to teach that concept directly.

Reply:
> On the other hand, I have been working with Elixir for 3 years and never had to worry about it much, so maybe it's too detailed for this exercise.

This is an excellent argument in favor of it being too detailed for this exercise I think.

> I think it could be a discussion point for the after.md document, but it also means that there might be another exercise which should occur after this which then maybe features a recursive or iterating through a string (akin to rna-transcription) to teach that concept directly.

Personally, I think the latter option where there is a separate exercise that could combine it with another concept like string iterating would work well!

Reply:
I like the idea of a latter concept that would focus on this topic. It should also probably touch regular expressions to teach about the unicode modifier - something many people struggle with on the current track, with `word_count`.

I added it to "out of scope".